ivy/functional/backends/tensorflow/experimental/layers.py

```diff
@@ -992,35 +1043,20 @@ def fft2(
 ) -> Union[tf.Tensor, tf.Variable]:
     if s is None:
         s = (x.shape[dim[0]], x.shape[dim[1]])
-    if all(j < -len(x.shape) for j in s):
-        raise ivy.utils.exceptions.IvyError(
-            f"Invalid dim {dim}, expecting ranging"
-            " from {-len(x.shape)} to {len(x.shape)-1}  "
-        )
-    if not all(isinstance(j, int) for j in s):
-        raise ivy.utils.exceptions.IvyError(
-            f"Expecting {s} to be a sequence of integers <class integer>"
-        )
-    if all(j <= 1 for j in s):
-        raise ivy.utils.exceptions.IvyError(
-            f"Invalid data points {s}, expecting s points larger than 1"
-        )
-    if norm != "backward" and norm != "ortho" and norm != "forward":
-        raise ivy.utils.exceptions.IvyError(f"Unrecognized normalization mode {norm}")
-    operation_name = f"{s} points FFT at dim {dim} with {norm} normalization"
-    if len(x.shape) == 2:
+    if len(x.shape) > 2:
+        result = _fft2_helper(x, s, dim)
+    else:
         x_new = trans_x_to_s(x, s, dim)
         x_complex = tf.cast(x_new, tf.complex128)
-        tf_fft2 = tf.signal.fft2d(x_complex, name=operation_name)
-    elif len(x.shape) > 2:
-        x_s = [trans_x_to_s(x[:, :, i], s, dim) for i in range(x.shape[2])]
-        x_new = tf.convert_to_tensor(x_s, dtype=x.dtype)
-        x_complex = tf.cast(x_new, tf.complex128)
-        tf_fft2 = tf.transpose(tf.signal.fft2d(x_complex, name=operation_name))
+        result = tf.signal.fft2d(x_complex)
+
+    result = _fft2_norm(result, s, dim, norm)
 
-    # Apply the same normalization as 'backward' in NumPy
-    tf_fft2 = _fft2_norm(tf_fft2, s, dim, norm)
-    return tf_fft2
+    if out is not None:
```

Comment:
Hi @AbdullahSabry,
`out` argument will be handled by `handle_out_argument` decorator. so, you no need to handle this. Please remove this block as it's reduntant.
Thanks